### PR TITLE
Remove scaffolds scss

### PR DESCRIPTION
**Pull request by AdamMusa**
**The rails d scaffold command "AppName" removes all generated application files except a file in scaffolds.scss which is in app/assets/stylesheets. I made an addition to remove scaffolds.scss
**

### CODE REVIEW

**Review by AdamMusa — COMMENTED** [review]
**Here is after launching the scaffold command. Take a good look at the scaffolds.scss file on the last line.**


before


**Here after destroying the scaffold command without my modifications. The destroy command does not delete the generated scaffolds.scss file, it just skip and I have checked in my directory hierarchy and put it there.**


with


**here after with my modifications. So I delete the scaffolds.scss file generated by the scaffold command**

after

**Comment by adityapandit17** [comment]
@AdamMusa Don't you think, it will delete a scaffolds.scss file for all previous scaffold. 
For example, 
`rails g scaffold post ` will generate a scaffold file.
again on 
`rails g scaffold article` it will not generate a scaffold file.
But after you changes when calling `rails d scaffold article` it will delete file for Post scaffold too.

**Comment by pixeltrix** [comment]
@AdamMusa thanks for your PR, however as @adityapandit17 points out we can't be certain that the scaffold.css file isn't being used by some elements in an unrelated template so we have to leave it in place. If you look at the tests for this feature you can see that we assert the file remains so it is intentional:

https://github.com/rails/rails/blob/main/railties/test/generators/scaffold_generator_test.rb#L224

Hope that clears things up - thanks again 👍🏻

**Comment by AdamMusa** [comment]
Good morning and thank you for this fast. Made the file generated by scaffold has css codes that gene the app for example black over , color on black button etc. @adityapandit17 precisely it does not delete after throwing rails d scaffold article. The specch scaffolds.scss will be spared by the precedente command.
Try you will see it that the scaffolds.scss file will be spared

**Comment by AdamMusa** [comment]
No, I'm not talking about withdrawing it for good @pixeltrix . When we run rails d scaffold it should delete all the file it just created. With my little line of code he will do it.

**Comment by pixeltrix** [comment]
@AdamMusa do you mean running it without an argument? e.g:

```
bundle exec rails destroy scaffold
```

If so that's not currently supported by the scaffold generator - it must have a model name. Additionally, where you've placed the method will result in the file being deleted for any destroy command - even ones unrelated to scaffold. There's a case for allowing it to run without an argument for removing that file but you'll need to alter the scaffold_generator.rb file - modifying the global destroy command isn't going to work.

I'll reopen the PR so you can work on it.

**Comment by AdamMusa** [comment]
Thanks @pixeltrix

File: railties/lib/rails/commands/destroy/destroy_command.rb [hunk 1]
```diff
@@ -14,14 +15,21 @@ def help
         end
       end
 
+      def delete_css_file_generate_with_scaffold
+        path = Rails.root.join('app', 'assets', 'stylesheets', 'scaffolds.scss')
+        FileUtils.remove_file(path,force=true)
+        puts " "*6+"\e[31mremove\e[0m"+" "*4  + path.to_s.split("/").reverse.slice(0,4).reverse.join("/")
```
Comment (zzak) [inline railties/lib/rails/commands/destroy/destroy_command.rb:21]: This feels like we're doing more work here than we need to, a couple of suggestions:

* Make sure there isn't a better place for this helper function
* Separate the `FileUtils.remove_file` logic from the `puts <massive-string-expression>`
* Add tests for `<massive-string-expression>`
* Add a railtie test to cover this usage

I think we'll need to see those to feel confident about adding this code to Rails.

File: Gemfile.lock [hunk 2]
```diff
@@ -642,4 +636,4 @@ DEPENDENCIES
   websocket-client-simple!
 
 BUNDLED WITH
-   2.2.15
+   2.2.16
```
[the remark this answers is not in the document]
Comment (AdamMusa) [reply Gemfile.lock:639]: i have just deleted. when i will finish to resolve the probleme by the good way i'm going to push the modification.

**Comment by zzak** [comment]
@AdamMusa Could you please rebase the `Gemfile.lock` changes out? :bow:

**Review by AdamMusa — COMMENTED** [review]
rebase Gemfile.lock

**Comment by AdamMusa** [comment]
@zzak it's ok?

File: Gemfile.lock [hunk 3]
```diff
@@ -634,4 +636,4 @@ DEPENDENCIES
   websocket-client-simple!
 
 BUNDLED WITH
-   2.2.16
+   2.2.15
```
Comment (zzak) [inline Gemfile.lock:639]: @AdamMusa I particularly meant this whole file, can you remove these changes from the PR?

**Pull request closed without merging**